lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3InputFile.java

```diff
@@ -109,15 +109,20 @@ private GetObjectRequest newGetObjectRequest()
     private boolean headObject()
             throws IOException
     {
-        HeadObjectRequest request = HeadObjectRequest.builder()
+        HeadObjectRequest.Builder request = HeadObjectRequest.builder()
                 .overrideConfiguration(context::applyCredentialProviderOverride)
                 .requestPayer(requestPayer)
                 .bucket(location.bucket())
-                .key(location.key())
-                .build();
+                .key(location.key());
+
+        if (context.sseType().equals(S3FileSystemConfig.S3SseType.CUSTOMER)) {
```

Comment:
can we reuse `S3Input::addEncryptionSettings`?

Reply:
addressed